lucene/core/src/java/org/apache/lucene/codecs/lucene90/Lucene90VectorReader.java

```diff
@@ -387,8 +387,40 @@ public int nextDoc() {
 
     @Override
     public int advance(int target) throws IOException {
-      // We could do better by log-binary search in ordToDoc, but this is never used
-      return slowAdvance(target);
+      assert docID() < target;
+      int idx = binarySearch(target);
```

Comment:
I wonder if we can simply use `Arrays.binarySearch(fieldEntry.ordToDoc, ord + 1 ,fieldEntry.ordToDoc.length - 1, target);` here?